addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -687,15 +687,21 @@ def _l10n_vn_edi_add_item_information(self, json_values):
         }
         for line in self.invoice_line_ids.filtered(lambda ln: ln.display_type == 'product'):
             # For credit notes amount, we send negative values (reduces the amount of the original invoice)
+            price_unit = line.price_unit
+            if line.tax_ids and line.tax_ids[0].price_include:
+                res = line.tax_ids[0].compute_all(
+                    price_unit, product=line.product_id, partner=self.env['res.partner']
+                )
+                price_unit = res['total_excluded']
```

Comment:
We avoid using compute_all today as its computation is not perfect and can lead to rounding issues in some specific cases.
But anyway, the taxes do not matter much in this case. 
What we care about is to know how we arrived at the subtotal (which already excludes the tax-included taxes), so we can reverse the calculation.

```suggestion
            if line.quantity and line.discount != 100:
                price_subtotal_before_discount = line.price_subtotal / (1 - (line.discount / 100))
                price_unit = line.currency_id.round(price_subtotal_before_discount / line.quantity)
            else:
                price_unit = line.price_unit
```

This should handle the issue :smile:

Reply:
P/s: there is a way but i feel like it not correct is that we force validation 0 by passing it to sinvoice (sinvoice has parameter validation 0 is to ignore some validation like the example above), i use this approach for my company but i'm not like this way so much :v:

Reply:
Wait , your approach will return string, so it will wrong when multiply with -1. Althought it works with normal invoice

Reply:
I think we should follow the documetation that we need float value here and i think we should only do this when we have tax with price include, because some customer might wonder "why my price unit so many decimal" . Here is how the invoice look like
![image](https://github.com/user-attachments/assets/12f8c38b-1af2-4dd1-aa2f-4f51f036c74e)

And note this: if in your sinvoice website, if you set the price unit decimal to 6 then it will work when you pass as 6 decimal, if you set less than that, it will error too you know